R/correctionsataglance-data.R

```diff
@@ -1,357 +1,539 @@
     }
-    startSeq <- c(startD, startSeq)
   }
-
+  
   if(firstOfMonth_end){
-    endSeq <- startSeq[-1]
     startSeq <- head(startSeq, -1)
-  } else {
-    endSeq <- c(startSeq[-1], endD)
   }
   
-#   #don't print Month Year in plot if there isn't enough room inside the rectangle
-  dateSeq <- startSeq
-  realSeq <- dateSeq #so we're not passing NA into other places of the report
-  labelSeq <- format(dateSeq, " %m/%Y ")
-  for (i in 1:length(dateSeq)) { 
-    if (isTextLong(labelText=labelSeq[i],dateLim=NULL,startD=startSeq[i],endD=endSeq[i],totalDays=numdays))
-      dateSeq[i] <- NA
-    } 
-  
-  return(list(dateRange = c(startD, endD),
-              dateSeq = dateSeq,
-              realSeq = realSeq,
-              startMonths = startSeq,
-              endMonths = endSeq,
-              middleDate = median(startSeq)))
+  return(startSeq)
 }
 
-formatDataList <- function(dataIn, type, timezone, ...){
-  
-  args <- list(...)
-  
-  if(length(dataIn) == 0){
-    return()
-  }
-  
-  start_i <- which(names(dataIn) %in% c('startTime', 'startDate'))
-  end_i <- which(names(dataIn) %in% c('endTime', 'endDate'))
-  dataIn[[start_i]] <- flexibleTimeParse(dataIn[[start_i]], timezone)
-  dataIn[[end_i]] <- flexibleTimeParse(dataIn[[end_i]], timezone)
-  
-  if (!isEmptyOrBlank(type)) {
-    if(type == 'APPROVALS' && length(dataIn)>0){
-      for (i in 1:length(dataIn[[end_i]])) {
-        endT <- dataIn[[end_i]][i]
-        if (endT > as.Date("2100-12-31")) {
-          dataIn[[end_i]][i] <- toEndOfTime(endT)
-        }
-      }
-    }
-  }
-
-  if(!type %in% c('APPROVALS', 'META')){
-    type_i <- which(dataIn$processingOrder == type)
-    i <- type_i[order(dataIn[[start_i]][type_i])] #order by start date
-  } else {
-    i <- order(dataIn[[start_i]]) #order by start date
-  }
-
-  typeData <- list(startDates = dataIn[[start_i]][i],
-                   endDates = dataIn[[end_i]][i],
-                   dataNum = length(dataIn[[start_i]][i]))
-  
-  if(type == 'APPROVALS'){
-    approvalColors <- getApprovalColors(dataIn$level)
-    extraData <- list(apprCol = approvalColors,
-                      apprType = paste("Approval:", dataIn$description),
-                      apprDates = args$datesRange)
-  } else if(type == 'META') {
-    extraData <- list(metaLabel = dataIn[[args$annotation]][i])
-  } else {
-    extraData <- list(corrLabel = dataIn$type[i],
-                      applyDates = dataIn$appliedTimeUtc[i])
-  }
-  
-  typeData <- append(typeData, extraData)
-  
-  if(typeData$dataNum == 0){
-    typeData <- NULL
-  }
-
-  return(typeData)
+#' Get a list of end dates
+#' @description Given a startSeq date list and end date and whether 
+#' or not the end date is the first of the month, provides a list of 
+#' YYYY-MM-DD used in other functions
+#' @param startSeq the start date
+#' @param endD the end date
+#' @param firstOfMonth_end TRUE or FALSE whether the end date is the first of the month
```

Comment:
@lindsaycarr Whoops, no that's leftover from before I refactored it. There might actually be several leftover param tags that aren't relevant anymore. I didn't think to look for tags to remove, only tags that needed to be added. I'll take a look around for more